t/file.t

```diff
@@ -97,8 +97,24 @@ SKIP: {
 subtest "Regression Tests" => sub {
 
   # Regression test for https://github.com/libwww-perl/URI/issues/102
-  my $with_hashes = URI::file->new_abs("/tmp/###");
-  is($with_hashes, 'file:///tmp/%23%23%23', "issue GH#102");
+  {
+    my $with_hashes = URI::file->new_abs("/tmp/###");
+    is($with_hashes, 'file:///tmp/%23%23%23', "issue GH#102");
+  }
+
+  # URI 5.11 introduced a bug where URI::file could return the current
+  # working directory instead of the path defined.
+  # The bug was caused by a wrong quantifier in a regular expression in
+  # URI::_fix_uric_escape_for_host_part() which returned an empty string for
+  # all URIs that needed escaping ('%xx') but did not have a host part.
+  # The empty string in turn caused URI::file->new_abs() to use the current
+  # working directory as a default.
+  {
+    my $file_path   = URI::file->new_abs('/a/path/that/pretty likely/does/not/exist-yie1Ahgh0Ohlahqueirequ0iebu8ip')->file();
+    my $current_dir = URI::file->new_abs()->file();
+
+    isnt( $file_path, $current_dir, "URI $URI::VERSION exposes a bug that could leak or destroy files on your system ($^O)." );
```

Comment:
```suggestion
    isnt( $file_path, $current_dir, 'regression test for #102' );
```

The test output for this is:

```bash
ok 2 - URI 5.13 exposes a bug that could leak or destroy files on your system (darwin).
```

The test as it stands is a bit confusing. At first glance it sounds as if the current version of URI has a dangerous bug. I think it would be easiest to have a simple test label.

Reply:
Agreed.